client/src/controllers/SessionController.ts

```diff
@@ -175,6 +167,12 @@ export class SessionController extends Controller<HTMLElement> {
     // workflow action modal) after the user confirms the dialog.
     if (!this.interceptValue || !this.hasWDialogOutlet) return;
 
+    // If the action button is inside a dialog, we need to hide the dialog first
+    // so it doesn't interfere with the confirmation dialog
+    this.lastActionButton
+      ?.closest('[data-controller="w-dialog"]')
+      ?.dispatchEvent(new Event('w-dialog:hide'));
```

Comment:
We can also show the dialog again after the overwrite has been confirmed by dispatching the `w-dialog:show` event in `confirmAction`, but I opted not to as per our discussion.